resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/config.scala

```diff
@@ -380,30 +380,62 @@ package object config extends Logging {
         " resource staging server to download files.")
       .internal()
       .stringConf
-      .createWithDefault(INIT_CONTAINER_DOWNLOAD_FILES_SECRET_PATH)
+      .createWithDefault(INIT_CONTAINER_SUBMITTED_FILES_DOWNLOAD_FILES_SECRET_PATH)
+
+  private[spark] val INIT_CONTAINER_REMOTE_JARS =
+    ConfigBuilder("spark.kubernetes.driver.initcontainer.remoteJars")
+      .doc("Comma-separated list of jar URIs to download in the init-container. This is inferred" +
+        " from spark.jars.")
+      .internal()
+      .stringConf
+      .createOptional
+
+  private[spark] val INIT_CONTAINER_REMOTE_FILES =
+    ConfigBuilder("spark.kubernetes.driver.initcontainer.remoteFiles")
+      .doc("Comma-separated list of file URIs to download in the init-container. This is inferred" +
+        " from spark.files.")
+      .internal()
+      .stringConf
+      .createOptional
 
   private[spark] val INIT_CONTAINER_DOCKER_IMAGE =
     ConfigBuilder("spark.kubernetes.driver.initcontainer.docker.image")
       .doc("Image for the driver's init-container that downloads mounted dependencies.")
       .stringConf
       .createWithDefault(s"spark-driver-init:$sparkVersion")
 
-  private[spark] val DRIVER_LOCAL_JARS_DOWNLOAD_LOCATION =
-    ConfigBuilder("spark.kubernetes.driver.mountdependencies.jarsDownloadDir")
+  private[spark] val DRIVER_SUBMITTED_JARS_DOWNLOAD_LOCATION =
+    ConfigBuilder("spark.kubernetes.driver.mountdependencies.submittedJars.downloadDir")
```

Comment:
We probably want to indicate that this is where the jars are being downloaded to. `spark.kubernetes.driver.jars` is ambiguous in the sense that it could be jars that need to be uploaded or downloaded or added to its classpath, etc.